src/main/java/com/amazon/opendistroforelasticsearch/jdbc/types/TimestampType.java

```diff
@@ -73,6 +74,19 @@ private Timestamp localDateTimeToTimestamp(LocalDateTime ldt, Calendar calendar)
             if (value.length() > 11 && value.charAt(10) == 'T') {
                 value = value.replace('T', ' ');
             }
+            // Timestamp.valueOf() does not like timezone information
+            if (value.length() > 23) {
+                if(value.length() == 24 && value.charAt(23) == 'Z') {
+                    value = value.substring(0, 23);
+                }
+                else if(value.charAt(23) == '+' || value.charAt(23) == '-') {
+                    // 'calendar' parameter takes precedence
+                    if(calendar==null) {
```

Comment:
NP: Please also add a space around the ```==``` here and around the ```+```  for ```"GMT"+value.substring(23))``` for style consistency.

Reply:
ok, will do